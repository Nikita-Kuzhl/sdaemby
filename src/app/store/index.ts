import { authReducer } from './../features/auth/index'
import { authApi } from './../service/authService'
import { paginationReducer } from './../features/pagination/index'
import { newsApi } from './../service/newsService'
import { configureStore } from '@reduxjs/toolkit'

export const store = configureStore({
  reducer: {
    [newsApi.reducerPath]: newsApi.reducer,
    [authApi.reducerPath]: authApi.reducer,
    pagination: paginationReducer,
    auth: authReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(newsApi.middleware, authApi.middleware),
})
export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch
