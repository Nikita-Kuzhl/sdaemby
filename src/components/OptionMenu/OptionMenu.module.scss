.container {
  position: relative;
  // height: 314px;
  width: 100%;
  border-top: 1px solid rgba(78, 100, 249, 0.1);
  background: #fff;
  box-shadow: 0px 20px 40px rgba(135, 124, 202, 0.15);
  padding-bottom: 30px;
}
.main {
  width: 1240px;
  margin: 0 auto;
  padding-top: 20px;
}
.tag__list {
  display: flex;
  gap: 48px;
}

.tag__title {
  font-weight: 500;
  font-size: 14px;
  line-height: 17px;
  color: #664ef9;
}
.select {
  margin-top: 11px;
  width: 200px;
  height: 37px;
}
.check__list {
  margin-top: 30px;
  display: flex;
  flex-wrap: wrap;
  gap: 50px;
  row-gap: 10px;
}
.check__item {
  display: flex;
  align-items: center;
  width: 200px;
  gap: 10px;
}
.check__title {
  font-weight: 400;
  font-size: 14px;
  line-height: 17px;
  color: #1e2123;
}
.border__bottom {
  border-bottom: 3px solid #4e64f9;
}
