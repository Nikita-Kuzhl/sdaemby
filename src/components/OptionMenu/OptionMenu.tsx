import React, { FC } from 'react'
import Select from 'react-select'
import { useGetAreaQuery, useGetMetroQuery } from '../../app/service/apartService'
import { ISelectOnChangePropsNumber, ISelectOnChangePropsString } from '../../app/types/selectTypes'
import { checkedApartParams, sleepingPlaces as sleepingPlacesList } from '../../constant'
import CheckBox from '../CheckBox'
import SelectStyle from '../SelectStyle'
import styles from './OptionMenu.module.scss'

interface IParams {
  city?: number
  rooms: number
  priceTo: number
  priceFrom: number
  sleepingPlaces: string
  area: number
  metro: number
  checkbox: string[]
}

interface IProps {
  cityId: number
  param: IParams
  setParam: React.Dispatch<React.SetStateAction<IParams>>
  isMainPage?: boolean
}
const OptionMenu: FC<IProps> = ({ cityId, param, setParam, isMainPage = false }) => {
  const { data: metroData, isSuccess: isSuccessMetro } = useGetMetroQuery(cityId)
  const { data: areaData, isSuccess: isSuccessArea } = useGetAreaQuery(cityId)

  return (
    <section
      className={styles.container}
      style={isMainPage ? { width: 1060, borderRadius: '0 0 10px 10px' } : {}}
    >
      <div className={styles.main} style={isMainPage ? { width: 1060, paddingLeft: 20 } : {}}>
        <ul className={styles.tag__list}>
          <div className={styles.tag__item}>
            <p className={styles.tag__title}>Спальные места</p>
            <Select
              placeholder='Выберите'
              styles={SelectStyle}
              options={sleepingPlacesList}
              value={
                param.sleepingPlaces
                  ? sleepingPlacesList.filter((i) => i.value === param.sleepingPlaces)[0]
                  : ''
              }
              // eslint-disable-next-line
              //@ts-ignore
              onChange={(e: ISelectOnChangePropsString) =>
                setParam({ ...param, sleepingPlaces: e.value })
              }
              className={styles.select}
            />
          </div>
          {!isMainPage && (
            <>
              <div className={styles.tag__item}>
                <p className={styles.tag__title}>Район</p>
                <Select
                  placeholder='Выберите'
                  styles={SelectStyle}
                  value={
                    isSuccessArea && param.area
                      ? areaData
                          .map((i) => {
                            if (i.id === param.area) {
                              return { value: i.id, label: i.name }
                            }
                          })
                          .filter((i) => i !== undefined)[0]
                      : ''
                  }
                  options={
                    isSuccessArea && [
                      ...areaData.map((item) => {
                        return { value: item.id, label: item.name }
                      }),
                    ]
                  }
                  // eslint-disable-next-line
                  //@ts-ignore
                  onChange={(e: ISelectOnChangePropsNumber) =>
                    setParam({ ...param, area: e.value })
                  }
                  className={styles.select}
                />
              </div>
              <div className={styles.tag__item}>
                <p className={styles.tag__title}>Метро</p>
                <Select
                  placeholder='Выберите'
                  styles={SelectStyle}
                  value={
                    isSuccessMetro && param.metro
                      ? metroData
                          .map((i) => {
                            if (i.id === param.metro) {
                              return { value: i.id, label: i.name }
                            }
                          })
                          .filter((i) => i !== undefined)[0]
                      : ''
                  }
                  options={
                    isSuccessMetro && [
                      ...metroData.map((item) => {
                        return { value: item.id, label: item.name }
                      }),
                    ]
                  }
                  // eslint-disable-next-line
                  //@ts-ignore
                  onChange={(e: ISelectOnChangePropsNumber) =>
                    setParam({ ...param, metro: e.value })
                  }
                  className={styles.select}
                />
              </div>
            </>
          )}
        </ul>
        <ul className={styles.check__list}>
          {checkedApartParams.map((i) => (
            <button
              onClick={() =>
                param.checkbox.includes(i.value)
                  ? setParam({
                      ...param,
                      checkbox: param.checkbox.filter((item) => item !== i.value),
                    })
                  : setParam({ ...param, checkbox: [...param.checkbox, i.value] })
              }
              key={i.id}
              className={styles.check__item}
            >
              <CheckBox active={param.checkbox.includes(i.value)} />
              <p className={styles.check__title}>{i.label}</p>
            </button>
          ))}
        </ul>
      </div>
    </section>
  )
}

export default OptionMenu
