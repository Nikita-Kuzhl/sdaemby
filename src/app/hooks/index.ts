import { AppDispatch, RootState } from '../store'
import axios from 'axios'
import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux'

export const useAppDispatch = () => useDispatch<AppDispatch>()
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector
export const api = axios.create({
  baseURL: process.env.REACT_APP_API,
  headers: {
    'Content-Type': 'application/json',
  },
})
