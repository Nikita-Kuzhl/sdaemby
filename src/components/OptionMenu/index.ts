import OptionMenu from './OptionMenu'

export default OptionMenu
