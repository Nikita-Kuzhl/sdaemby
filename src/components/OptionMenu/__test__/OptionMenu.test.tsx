import { render, screen } from '@testing-library/react'
import { renderHook } from '@testing-library/react-hooks'
import userEvent from '@testing-library/user-event'

import { useCallback, useState } from 'react'
import { Provider } from 'react-redux'
import { BrowserRouter } from 'react-router-dom'
import { store } from '../../../app/store'
import OptionMenu from '../OptionMenu'

const useParam = (value) => {
  const [param, setParam] = useState(value)
  const reset = useCallback(() => setParam(value), [value])
  return { param, reset }
}
test('render option menu', async () => {
  const paramValue = {
    rooms: 1,
    priceTo: 1,
    priceFrom: 1,
    sleepingPlaces: '2+1',
    area: 1,
    metro: 1,
    checkbox: [],
  }
  const hook = renderHook(() => useParam(paramValue))
  await render(
    <Provider store={store}>
      <BrowserRouter>
        <OptionMenu
          param={hook.result.current.param}
          setParam={hook.result.current.reset}
          cityId={1}
        />
      </BrowserRouter>
    </Provider>,
  )
  expect(screen.getAllByTestId('option menu')).toMatchSnapshot()
})
export {}
