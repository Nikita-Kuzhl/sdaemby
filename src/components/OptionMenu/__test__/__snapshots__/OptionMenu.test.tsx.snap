// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render option menu 1`] = `
Array [
  <section
    class="container"
    data-testid="option menu"
  >
    <div
      class="main"
    >
      <ul
        class="tag__list"
      >
        <form
          class="tag__item"
          data-tesid="select sleeping places"
        >
          <label
            class="tag__title"
            for="sleeping places"
          >
            Спальные места
          </label>
          <div
            class="select css-b62m3t-container"
          >
            <span
              class="css-1f43avz-a11yText-A11yText"
              id="react-select-2-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="css-1f43avz-a11yText-A11yText"
            />
            <div
              class=" css-7q3fjo-control"
            >
              <div
                class=" css-fn5jfe-Component"
              >
                <div
                  class=" css-qc6sy-singleValue"
                >
                  2+1
                </div>
                <div
                  class=" css-1u4kn00-Component"
                  data-value=""
                >
                  <input
                    aria-autocomplete="list"
                    aria-expanded="false"
                    aria-haspopup="true"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    class=""
                    id="sleeping places"
                    role="combobox"
                    spellcheck="false"
                    style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class=" css-ny0e4k-Component"
              >
                <span
                  class=" css-ds71as-Component"
                />
                <div
                  aria-hidden="true"
                  class=" css-pkfgqp-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-tj5bde-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <input
              name="sleeping places"
              type="hidden"
              value="2+1"
            />
          </div>
        </form>
        <div
          class="tag__item"
        >
          <p
            class="tag__title"
          >
            Район
          </p>
          <div
            class="select css-b62m3t-container"
          >
            <span
              class="css-1f43avz-a11yText-A11yText"
              id="react-select-3-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="css-1f43avz-a11yText-A11yText"
            />
            <div
              class=" css-7q3fjo-control"
            >
              <div
                class=" css-fn5jfe-Component"
              >
                <div
                  class=" css-gxrbjt-placeholder"
                  id="react-select-3-placeholder"
                >
                  Выберите
                </div>
                <div
                  class=" css-1u4kn00-Component"
                  data-value=""
                >
                  <input
                    aria-autocomplete="list"
                    aria-describedby="react-select-3-placeholder"
                    aria-expanded="false"
                    aria-haspopup="true"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    class=""
                    id="react-select-3-input"
                    role="combobox"
                    spellcheck="false"
                    style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class=" css-ny0e4k-Component"
              >
                <span
                  class=" css-ds71as-Component"
                />
                <div
                  aria-hidden="true"
                  class=" css-pkfgqp-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-tj5bde-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="tag__item"
        >
          <p
            class="tag__title"
          >
            Метро
          </p>
          <div
            class="select css-b62m3t-container"
          >
            <span
              class="css-1f43avz-a11yText-A11yText"
              id="react-select-4-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="css-1f43avz-a11yText-A11yText"
            />
            <div
              class=" css-7q3fjo-control"
            >
              <div
                class=" css-fn5jfe-Component"
              >
                <div
                  class=" css-gxrbjt-placeholder"
                  id="react-select-4-placeholder"
                >
                  Выберите
                </div>
                <div
                  class=" css-1u4kn00-Component"
                  data-value=""
                >
                  <input
                    aria-autocomplete="list"
                    aria-describedby="react-select-4-placeholder"
                    aria-expanded="false"
                    aria-haspopup="true"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    class=""
                    id="react-select-4-input"
                    role="combobox"
                    spellcheck="false"
                    style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class=" css-ny0e4k-Component"
              >
                <span
                  class=" css-ds71as-Component"
                />
                <div
                  aria-hidden="true"
                  class=" css-pkfgqp-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-tj5bde-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </ul>
      <ul
        class="check__list"
      >
        <button
          class="check__item"
        >
          <section
            class="container"
            data-testid="checkbox"
          >
            <img
              src="/assets/images/check.svg"
            />
          </section>
          <p
            class="check__title"
          >
            Газовая плита
          </p>
        </button>
        <button
          class="check__item"
        >
          <section
            class="container"
            data-testid="checkbox"
          >
            <img
              src="/assets/images/check.svg"
            />
          </section>
          <p
            class="check__title"
          >
            Духовка
          </p>
        </button>
        <button
          class="check__item"
        >
          <section
            class="container"
            data-testid="checkbox"
          >
            <img
              src="/assets/images/check.svg"
            />
          </section>
          <p
            class="check__title"
          >
            Кофеварка
          </p>
        </button>
        <button
          class="check__item"
        >
          <section
            class="container"
            data-testid="checkbox"
          >
            <img
              src="/assets/images/check.svg"
            />
          </section>
          <p
            class="check__title"
          >
            Микроволновая печь
          </p>
        </button>
        <button
          class="check__item"
        >
          <section
            class="container"
            data-testid="checkbox"
          >
            <img
              src="/assets/images/check.svg"
            />
          </section>
          <p
            class="check__title"
          >
            Посуда
          </p>
        </button>
        <button
          class="check__item"
        >
          <section
            class="container"
            data-testid="checkbox"
          >
            <img
              src="/assets/images/check.svg"
            />
          </section>
          <p
            class="check__title"
          >
            Посудочная машина
          </p>
        </button>
      </ul>
    </div>
  </section>,
]
`;
